### ISSUE TITLE
24%+15%=100%? (Unpainted areas)

### ISSUE THREAD

**Issue by DerLalo** [nekromoff/arrogance-of-space#16]
I used this tool for a street in my neighbourhood. I took a few meters and painted them red and blue. The created image says: Cars (24%) Pedestrians (15%). What about the rest?

![Aufstieg_arrogance](https://github.com/nekromoff/arrogance-of-space/assets/137991233/2558cbde-2b83-49f8-83b2-15ba25ac9763)

If I paint all the rest black, the result is Cars (11%) Pedestrians (7%) Dead space (82%).
![Aufstieg_arrogance2](https://github.com/nekromoff/arrogance-of-space/assets/137991233/f4d6d3aa-1434-4010-ab59-79995ea65a08)

So the must be an issue with unpainted areas! In my opinion, they should either be ignored, so the result for the first image should be Cars (62%) Pedestrians (38%). Or they should be handled as dead space, and the result would be the same as in the second example.

**Comment by nekromoff** [comment]
Indeed, this is the case. All squares are counted towards 100%, but if some
are left unmarked, percentage is not calculated for them.

Automatic marking of everything unmarked as dead space probably wouldn't be
correct. So for now I would leave it to a user to decide.

I am leaving this issue open, if somebody else can come with more ideas on
how to address this.

**Comment by DerLalo** [comment]
OK, I found the bug: it's the eraser. I uploaded an empty, square picture, scaled to fit 10x10 tiles. In the first step, I marked 6 lines red. The result is correct, 100% Cars:
![square1](https://github.com/nekromoff/arrogance-of-space/assets/137991233/387dff73-2372-472d-986b-6ad0e7c0e2eb)

But if I delete one line, the result is 82% Cars:
![square2](https://github.com/nekromoff/arrogance-of-space/assets/137991233/cf846c2a-2820-4086-92d0-550bf1dd986b)

So the erased tiles are still counted and not handled as untouched tiles!

**Comment by nekromoff** [comment]
Good catch! I might fix it at some point. Until then don't use the eraser
tool  :-)

**Comment by nekromoff** [comment]
fixed
